### DISCUSSION TITLE
Where is the function `open` defined?

### DISCUSSION THREAD

**Discussion by ghost** [dbsoft/dwindows#12] [General]
I don't understand this piece of code on `dw.h`, please elaborate more @dbsoft 

```
#if defined(__ANDROID__)
int API dw_file_open(const char *path, int mode);
#else
#define dw_file_open(a, b) open(a, b)
#endif
```

**Comment by ghost** [comment]
It turned out `open` is a POSIX function and Microsoft also implemented it for Windows:

https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/open?view=msvc-170

**Comment by dbsoft** [comment]
Ok, open() is a system function... however special handling was required on Android... calling into Java.... so dw_file_open() is implemented to call into Java on Android and on all other platforms it calls the CRT function open().

If I remember correctly, just calling open() on Android used the Linux open() which did not have permissions to access the files, so they always failed.
